pkg/trace/event/processor.go

```diff
@@ -66,24 +67,26 @@ func (p *Processor) Process(root *pb.Span, t *pb.TraceChunk) (numEvents, numExtr
 		}
 
 		numExtracted++
-
-		sampled, epsRate := p.maxEPSSample(span, priority)
-		if !sampled {
-			continue
+		//event analytics tags shouldn't be set on sampled single spans
+		if !(span.Metrics != nil &&
+			span.Metrics[sampler.KeySpanSamplingMechanism] == float64(apmsampling.SpanSamplingRuleMechanism)) {
+			sampled, epsRate := p.maxEPSSample(span, priority)
+			if !sampled {
+				continue
+			}
+			sampler.SetMaxEPSRate(span, epsRate)
+			sampler.SetClientRate(span, clientSampleRate)
+			sampler.SetPreSampleRate(span, preSampleRate)
+			sampler.SetEventExtractionRate(span, extractionRate)
```

Comment:
I'd argue that these belong in the extractors, instead of here, but that change is not necessarily the responsibility of this PR.

Reply:
Have you considered this? Does it make sense? What I'm trying to say is that: all these tags are very specific to the individual extractors, and it feels as if it is the extractors that need to add them. I feel that having them here is just an (incorrect) way to remove duplication. In this case, it seems that having the duplication is a better separation of concerns.

What I'm saying is: these tags need to be set in the individual extractors. If needed, make a helper method and call it in all of them, rather than doing it here. I feel like this component shouldn't care about what type of extractors are running and shouldn't do such checks.

Reply:
I considered, but I don't see how that would make code clearer, as half of the tags depend on the root span data( client sample rate, preSampleRate) , or tracer (priority). And MaxEPSRate depends on the maxEPSSampler field of the processor struct.

Reply:
Ok, I see what you mean. That's fine then. Let's not do too much work here that is unrelated to the PR. Unless you think think that it's much better to do that, feel free to resolve this comment.